xgcm/test/test_transform.py

```diff
@@ -0,0 +1,724 @@
+    if error_flag:
+        with pytest.xfail():
+            transformed = grid.transform(source.data, axis, target, **transform_kwargs)
+    else:
+        transformed = grid.transform(source.data, axis, target, **transform_kwargs)
+        xr.testing.assert_allclose(transformed, expected.data)
+
+
+def test_grid_transform_auto_naming(all_cases):
+    """Check that the naming for the new dimension is adapted for the output if the target is not passed as xr.Dataarray"""
+    source, grid_kwargs, target, transform_kwargs, expected, error_flag = all_cases
+
+    # ? Not sure if we need to go through all the cases here...
+
+    axis = list(grid_kwargs["coords"].keys())[0]
+
+    grid = Grid(source, **grid_kwargs)
+
+    # modify the expected naming and convert target to numpy array
+    target_data = transform_kwargs.setdefault("target_data", None)
+
+    if transform_kwargs["target_data"] is None:
+        # When no target_data is provided default to axis coordinates depending on method
+        if transform_kwargs["method"] == "linear":
+            expected_data_coord = grid.axes[axis].coords["center"]
+        elif transform_kwargs["method"] == "conservative":
+            expected_data_coord = grid.axes[axis].coords["outer"]
+    else:
+        # When target_data is provided check against the name of the dataarray
+        expected_data_coord = target_data.name
+
+    target = target.data
+
+    if error_flag:
+        with pytest.xfail():
+            transformed = grid.transform(source.data, axis, target, **transform_kwargs)
+    else:
+        transformed = grid.transform(source.data, axis, target, **transform_kwargs)
+        assert expected_data_coord in transformed.coords
+
+
+""" Multidimensional tests with dask scheduler """
+
+
+# dask client fixtures (adapted from https://github.com/xgcm/fastjmd95/blob/2cbccf2144e73c13a91741029eb7fd273d380d28/fastjmd95/test/test_jmd95.py#L61)
+@pytest.fixture
+def no_client():
+    return None
+
+
+@pytest.fixture
+def threaded_client():
+    with dask.config.set(scheduler="threads"):
+        yield
+
+
+@pytest.fixture
+def processes_client():
+    with dask.config.set(scheduler="processes"):
+        yield
+
+
+@pytest.fixture(scope="module")
+def distributed_client():
+    from dask.distributed import Client, LocalCluster
+
+    cluster = LocalCluster(threads_per_worker=1, n_workers=2, processes=True)
+    client = Client(cluster)
+    yield
+    client.close()
+    del client
+    cluster.close()
+    del cluster
+
+
+all_clients = ["no_client", "threaded_client", "processes_client", "distributed_client"]
+
+
+@pytest.mark.parametrize("client", all_clients)
+def test_grid_transform_multidim(request, client, multidim_cases):
+    # broadcast the 1d column agains some other dims and make sure that the 1d results are still valid
+    source, grid_kwargs, target, transform_kwargs, expected, error_flag = multidim_cases
+
+    na = 3
+    source = source * xr.DataArray(np.ones([na]), dims=["a"])
```

Comment:
This may not be the optimal way to construct a multidimensional dataset, but I guess it works.

Reply:
Would love to learn how to do this better. I have been using this for a long time. On the upside, its very readable IMO.

Reply:
Not sure how to do this more elegantly. But would be happy to change this anytime if you have suggestions. For now its probably ok?